### PR TITLE
BAH-510 | Maharjun | fixed error on going to consultation tab from visit…

### PR DESCRIPTION
**Pull request by maharjunm**
… page or patient image

https://bahmni.atlassian.net/browse/BAH-510

### CODE REVIEW

File: ui/app/clinical/patientcontrolpanel/directives/patientControlPanel.js [hunk 1]
```diff
@@ -53,7 +59,7 @@ angular.module('bahmni.common.patient')
                     if ($scope.activeVisit) {
                         links.push({text: $translate.instant('CONTROL_PANEL_CONSULTATION_TEXT'), icon: "btn-consultation dashboard-btn", href: "#" + clinicalAppConfigService.getConsultationBoardLink()});
                     } else if (state.match("patient.visit")) {
-                        links.push({text: $translate.instant('CONTROL_PANEL_DASHBOARD_TEXT'), icon: "btn-summary dashboard-btn", href: "#/" + $stateParams.configName + "/patient/" + $scope.patient.uuid + "/dashboard"});
+                        links.push({text: "Dashboard", icon: "btn-summary dashboard-btn", href: $scope.getDashboardLink()});
```
Comment (sumanmaity112) [inline ui/app/clinical/patientcontrolpanel/directives/patientControlPanel.js]: Hi @maharjunm, wouldn't it require `internationalization`?

File: ui/test/unit/clinical/patientcontrolpanel/directives/patientControlPanel.spec.js [hunk 2]
```diff
@@ -254,4 +254,89 @@ describe('patientControlPanelTest', function () {
...
+            encounterUuid: "encounterUuid",
+            programUuid: "programUuid",
+            enrollment: "patientProgramUuid",
+            dateEnrolled: "dateEnrolled"
+        };
+
+        state = {current: {name : "patient.dashboard"}};
+
+        mockBackend.expectGET('patientcontrolpanel/views/controlPanel.html').respond("<div>dummy</div>");
+
```
Comment (sumanmaity112) [inline ui/test/unit/clinical/patientcontrolpanel/directives/patientControlPanel.spec.js]: @maharjunm, remove unwanted empty lines

File: ui/test/unit/clinical/patientcontrolpanel/directives/patientControlPanel.spec.js [hunk 3]
```diff
@@ -254,4 +254,89 @@ describe('patientControlPanelTest', function () {
...
+        scope.visitHistory={
+            activeVisit: true
+        };
+
+        scope.section = {
+            numberOfVisits:1
+        };
+        scope.patient = {uuid: "patientUuid"};
+
+
```
Comment (sumanmaity112) [inline ui/test/unit/clinical/patientcontrolpanel/directives/patientControlPanel.spec.js]: @maharjunm, remove unwanted empty lines

File: ui/test/unit/clinical/patientcontrolpanel/directives/patientControlPanel.spec.js [hunk 4]
```diff
@@ -254,4 +254,89 @@ describe('patientControlPanelTest', function () {
...
+        stateParams = {
+            configName: "default",
+            patientUuid: "patientUuid",
+            encounterUuid: "encounterUuid"
+        };
+
+        state = {current: {name : "patient.dashboard"}};
+
+        mockBackend.expectGET('patientcontrolpanel/views/controlPanel.html').respond("<div>dummy</div>");
+
```
Comment (sumanmaity112) [inline ui/test/unit/clinical/patientcontrolpanel/directives/patientControlPanel.spec.js]: @maharjunm, remove unwanted empty lines

File: ui/test/unit/clinical/patientcontrolpanel/directives/patientControlPanel.spec.js [hunk 5]
```diff
@@ -254,4 +254,89 @@ describe('patientControlPanelTest', function () {
...
+        var scope = rootScope.$new();
+        scope.visitHistory={
+            activeVisit: true
+        };
+
+        scope.section = {
+            numberOfVisits:1
+        };
+        scope.patient = {uuid: "patientUuid"};
+
```
Comment (sumanmaity112) [inline ui/test/unit/clinical/patientcontrolpanel/directives/patientControlPanel.spec.js]: @maharjunm, remove unwanted empty lines

File: ui/test/unit/clinical/patientcontrolpanel/directives/patientControlPanel.spec.js [hunk 6]
```diff
@@ -113,7 +113,7 @@ describe('patientControlPanelTest', function () {
         scope.$digest();
 
         expect(compiledElementScope.links).not.toBeUndefined();
-        expect(compiledElementScope.links).toEqual([{text: "CONTROL_PANEL_DASHBOARD_TEXT", icon: "btn-summary dashboard-btn", href: "#/default/patient/patientUuid/dashboard"}]);
+        expect(compiledElementScope.links).toEqual([{text: "Dashboard", icon: "btn-summary dashboard-btn", href: "#/default/patient/patientUuid/dashboard"}]);
```
Comment (djazayeri) [inline ui/test/unit/clinical/patientcontrolpanel/directives/patientControlPanel.spec.js]: i18n

**Comment by sumanmaity112** [comment]
Hi @maharjunm, 
Looks ok. Please the move the card to `Need code review` lane.

**Comment by sai789** [comment]
We moved this card to Code review.

**Comment by angshu** [comment]
@sai789 please follow the comments in card BAH-510

File: ui/app/clinical/patientcontrolpanel/views/controlPanel.html [hunk 7]
```diff
@@ -4,12 +4,12 @@
...
         <div class="patient-info">
             <div>
-                <a class="patient-img" href="{{getDashboardLink()}}" ng-click="changeContext($event)">
+                <a class="patient-img" href="{{::getDashboardLink()}}" ng-click="changeContext($event)">
                     <img class="patient-image" ng-src="{{::patient.image}}" fallback-src="../images/blank-user.gif">
                 </a>
                 <div class="patient-handler">
                     <h4 class="patient-name">
-                    <a href="{{getDashboardLink()}}" ng-click="changeContext($event)"><span bo-text="patient.name"></span></a><span ng-show="patient.genderText" ng-bind-html="' ('+patient.genderText+')'"></span></h4>
+                        <a href="{{::getDashboardLink()}}" ng-click="changeContext($event)"><span bo-text="patient.name"></span></a><span ng-show="patient.genderText" ng-bind-html="' ('+patient.genderText+')'"></span></h4>
```
Comment (sumanmaity112) [inline ui/app/clinical/patientcontrolpanel/views/controlPanel.html]: @maharjunm, indent it

File: ui/test/unit/clinical/patientcontrolpanel/directives/patientControlPanel.spec.js [hunk 8]
```diff
@@ -254,4 +254,66 @@ describe('patientControlPanelTest', function () {
 
         expect(compiledElementScope.isInEditEncounterMode()).toBeTruthy();
     })
+
+    it("goToDashBoard() should return the dashboardUrl with given program stateParams",function(){
```
Comment (sumanmaity112) [inline ui/test/unit/clinical/patientcontrolpanel/directives/patientControlPanel.spec.js]: @maharjunm, Indent the line. Also change **goToDashBoard()** to **goToDashBoard**

Comment (sai789) [reply ui/test/unit/clinical/patientcontrolpanel/directives/patientControlPanel.spec.js]: Done. Have a look at latest commit.

**Comment by sai789** [comment]
Hi @sumanmaity112 
We have addressed your feedback in the previous commit. Thanks

File: ui/test/unit/clinical/patientcontrolpanel/directives/patientControlPanel.spec.js [hunk 9]
```diff
@@ -253,5 +269,67 @@ describe('patientControlPanelTest', function () {
...
+        scope.patient = {uuid: "patientUuid"};
+        var element = compile(simpleHtml)(scope);
+        scope.$digest();
+        mockBackend.flush();
+        var compiledElementScope = element.isolateScope();
+        scope.$digest();
+        expect(compiledElementScope.getDashboardLink()).toEqual("#/programs/patient/patientUuid/dashboard?programUuid=programUuid&enrollment=patientProgramUuid&dateEnrolled=dateEnrolled");
+    });
+
+    it("goToDashBoard() should return the dashboardUrl with given default stateParams ", function () {
```
Comment (sumanmaity112) [inline ui/test/unit/clinical/patientcontrolpanel/directives/patientControlPanel.spec.js]: Hi @maharjunm, make `goToDashBoard()`  to `goToDashBoard
`

**Pull request merged**